src/cbuild/hooks/post_install/007_strip_debug.py

```diff
@@ -159,5 +161,59 @@ def invoke(pkg):
     else:
         (pkg.destdir / "usr/lib").rmdir()
 
+    # now collect the referenced debug sources
+    for v in ddest.rglob("*"):
+        if v.is_dir():
+            continue
+
+        for source_file in (
+            chroot.enter(
+                "llvm-dwarfdump",
+                "--show-sources",
+                f"/{v.relative_to(pkg.rparent.bldroot_path)}",
+                capture_output=True,
+                check=True,
+                ro_root=True,
+                ro_build=True,
+                unshare_all=True,
+            )
+            .stdout.strip()
+            .splitlines()
+        ):
+            source_file = pathlib.Path(source_file.strip().decode())
+
+            # don't have access to these!
+            if source_file.name in ["<stdin>"]:
+                continue
+
+            # /usr sourcefiles like /usr/include headers are already provided by -devel packages
+            if source_file.is_relative_to("/usr"):
+                continue
+
+            if not source_file.is_absolute():
+                src = pkg.rparent.srcdir / source_file
+
+                if not src.exists():
+                    # silence warning for crt objects
+                    if source_file.name not in [
+                        "Scrt1.c",
+                        "crt1.c",
+                        "crtbegin.c",
+                    ]:
+                        pkg.log_warn(
+                            f"missing debug source file: {source_file}"
+                        )
+                else:
+                    dst = (
+                        ddest
+                        / f"usr/src/debug/{pkg.pkgname}-{pkg.pkgver}-r{pkg.pkgrel}"
+                        / source_file
+                    )
+                    dst.parent.mkdir(parents=True, exist_ok=True)
+                    shutil.copy(src, dst)
+            else:
+                pkg.log_warn(f"unknown debug source file: {source_file}")
```

Comment:
it being absolute doesn't make it unknown, perhaps some other message is more interesting

Reply:
also forgot; how common is /usr/whatever in referenced debug files? we have a ton of potential warning spam here overall including the above (perhaps build like all the libs that gnome depends on and see what gets output now, etc), and it's the kind of warning spam that you can do nothing about (i.e. maybe is more useful to actually print nothing in these cases unless it is ever an accident that something can reference a /source path, and it was a legit thing that someone might want to fix; but i can't think of any, being silent wouldn't affect anything in that case)

Reply:
ok, seems the cargo case uses / for cache

```

0:00:16.947 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/ptr/span.rs
0:00:16.947 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/access.rs
0:00:16.948 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/gimli-0.29.0/src/read/str.rs
0:00:16.949 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/vec/ops.rs
0:00:16.949 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/crc-3.2.1/src/crc32/slice16.rs
0:00:16.949 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/addr2line-0.22.0/src/lazy.rs
0:00:16.950 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/gimli-0.29.0/src/common.rs
0:00:16.950 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/object-0.36.0/src/pod.rs
0:00:16.950 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/gimli-0.29.0/src/leb128.rs
0:00:16.950 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.86/src/wrapper.rs
0:00:16.951 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.86/src/ensure.rs
0:00:16.951 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/index.rs
0:00:16.951 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/object-0.36.0/src/read/read_ref.rs
0:00:16.952 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/gimli-0.29.0/src/read/rnglists.rs
0:00:16.952 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec_helpers-3.1.5/src/bitstream_io_impl/bitstream_io_reader.rs
0:00:16.952 => WARNING: libdovi-3.3.1-r0: unknown debug source file: /cbuild_cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.86/src/backtrace.rs
0:00:16.953 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/object-0.36.0/src/read/elf/section.rs
0:00:16.953 => WARNING: libdovi-3.3.1-r0: missing debug source file: vendor/object-0.36.0/src/read/mod.rs
```

Reply:
and answering my own question; a ton of builds have stuff like  

```
0:00:21.438 => WARNING: libnl-3.10.0-r0: missing debug source file: build/lib/route/cls/ematch_syntax.y
0:00:21.438 => WARNING: libnl-3.10.0-r0: missing debug source file: build/lib/route/cls/ematch_grammar.l
0:00:21.443 => WARNING: libnl-3.10.0-r0: missing debug source file: build/lib/route/pktloc_grammar.l
0:00:21.444 => WARNING: libnl-3.10.0-r0: missing debug source file: build/lib/route/pktloc_syntax.y
```

so just silence these

Reply:
> does that warrant another prefix remap?

a remap of what where how

Reply:
those paths exist and justwork if you didn't ignore-/ so i don't think anything has to be remapped

Reply:
hm actually no cause it would fail at runtime later

Reply:
hm yeah sure sounds good